### ISSUE TITLE
TF2.4 Build - An error occurred during the fetch of repository 'io_bazel_rules_go':

### ISSUE THREAD

**Issue by oracle3001** [tensorflow/tensorflow#44570]
**System information**
- OS Platform and Distribution : Windows 10
- TensorFlow installed from (source or binary): Source
- TensorFlow version: 2.4
- Python version: 3.7
- Bazel version (if compiling from source): 3.7.0
- GCC/Compiler version (if compiling from source): MSVC 2019
- CUDA/cuDNN version: Cuda 11.0 / cuDNN 8.0.4
- GPU model and memory: 2070 Max-Q

When building from source, I receive the following error when running the following command.

bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package

ERROR: An error occurred during the fetch of repository 'io_bazel_rules_go':
   Traceback (most recent call last):
        File "C:/users/adam/_bazel_adam/ahelnrbq/external/bazel_tools/tools/build_defs/repo/git.bzl", line 181, column 30, in _git_repository_implementation
                update = _clone_or_update(ctx)
        File "C:/users/adam/_bazel_adam/ahelnrbq/external/bazel_tools/tools/build_defs/repo/git.bzl", line 36, column 20, in _clone_or_update
                git_ = git_repo(ctx, directory)
        File "C:/users/adam/_bazel_adam/ahelnrbq/external/bazel_tools/tools/build_defs/repo/git_worker.bzl", line 91, column 12, in git_repo
                _update(ctx, git_repo)
        File "C:/users/adam/_bazel_adam/ahelnrbq/external/bazel_tools/tools/build_defs/repo/git_worker.bzl", line 101, column 9, in _update
                init(ctx, git_repo)
        File "C:/users/adam/_bazel_adam/ahelnrbq/external/bazel_tools/tools/build_defs/repo/git_worker.bzl", line 118, column 15, in init
                _error(ctx.name, cl, st.stderr)
        File "C:/users/adam/_bazel_adam/ahelnrbq/external/bazel_tools/tools/build_defs/repo/git_worker.bzl", line 190, column 9, in _error
                fail("error running '%s' while working with @%s:\n%s" % (command_text, name, stderr))
Error in fail: error running 'git init C:/users/adam/_bazel_adam/ahelnrbq/external/io_bazel_rules_go' while working with @io_bazel_rules_go:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("git" init C:/users/adam/_bazel_adam/ahelnrbq/external/io_bazel_rules_go): The system cannot find the file specified.
 (error: 2)
INFO: Repository com_google_protobuf instantiated at:
  C:/sdks/tensorflow-dev/WORKSPACE:19:16: in <toplevel>
  C:/sdks/tensorflow-dev/tensorflow/workspace.bzl:585:20: in tf_repositories
Repository rule tf_http_archive defined at:
  C:/sdks/tensorflow-dev/third_party/repo.bzl:131:34: in <toplevel>
INFO: Repository envoy_api instantiated at:
  C:/sdks/tensorflow-dev/WORKSPACE:116:10: in <toplevel>
  C:/users/adam/_bazel_adam/ahelnrbq/external/com_github_grpc_grpc/bazel/grpc_deps.bzl:235:21: in grpc_deps
Repository rule http_archive defined at:
  C:/users/adam/_bazel_adam/ahelnrbq/external/bazel_tools/tools/build_defs/repo/http.bzl:336:31: in <toplevel>
ERROR: no such package '@io_bazel_rules_go//go': error running 'git init C:/users/adam/_bazel_adam/ahelnrbq/external/io_bazel_rules_go' while working with @io_bazel_rules_go:
java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("git" init C:/users/adam/_bazel_adam/ahelnrbq/external/io_bazel_rules_go): The system cannot find the file specified.
 (error: 2)
INFO: Elapsed time: 28.671s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
    Fetching @upb; fetching
    Fetching @build_bazel_rules_apple; fetching
    Fetching @build_bazel_apple_support; fetching
    Fetching ...al/upb; Extracting C:/users/adam/_bazel_adam/ahelnrbq/external/upb/temp630520761\
6925763744/9effcbcb27f0a665f9f345030188c0b291e32482.tar.gz
    Fetching ..._apple; Extracting C:/users/adam/_bazel_adam/ahelnrbq/external/build_bazel_rules\
_apple/temp16001111932413110489/5131f3d46794bf227d296c82f30c2499c9de3c5b.tar.gz
    Fetching ...upport; Extracting C:/users/adam/_bazel_adam/ahelnrbq/external/build_bazel_apple\
_support/temp14862261056236700618/501b4afb27745c4813a88ffa28acd901408014e4.tar.gz

**Comment by Saduf2019** [comment]
@oracle3001 
Please share the steps you ran before you encountered the error.
and please to follow [this guide](https://www.tensorflow.org/install/source_windows) 

def minute_job():
t = time.localtime()
current_time = time.strftime("%H:%M:%S", t)
print("From minute job:", current_time)
minute_job()

**Comment by oracle3001** [comment]
I was following the guide.

Pulled 2.4 branch from github.

Ran `python ./configure.py`, used all the defaults i.e.

```
You have bazel 3.1.0 installed.
Please specify the location of python. [Default is C:\Users\Adam\anaconda3\python.exe]:

Found possible Python library paths:
  C:\Users\Adam\anaconda3\lib\site-packages
Please input the desired Python library path to use.  Default is [C:\Users\Adam\anaconda3\lib\site-packages]

Do you wish to build TensorFlow with ROCm support? [y/N]:
No ROCm support will be enabled for TensorFlow.

Do you wish to build TensorFlow with CUDA support? [y/N]:
No CUDA support will be enabled for TensorFlow.

Please specify optimization flags to use during compilation when bazel option "--config=opt" is specified [Default is /arch:AVX]:


Would you like to override eigen strong inline for some C++ compilation to reduce the compilation time? [Y/n]:
Eigen strong inline overridden.

Would you like to interactively configure ./WORKSPACE for Android builds? [y/N]:
Not configuring the WORKSPACE for Android builds.

Preconfigured Bazel build configs. You can use any of the below by adding "--config=<>" to your build command. See .bazelrc for more details.
        --config=mkl            # Build with MKL support.
        --config=mkl_aarch64    # Build with oneDNN support for Aarch64.
        --config=monolithic     # Config for mostly static monolithic build.
        --config=numa           # Build with NUMA support.
        --config=dynamic_kernels        # (Experimental) Build kernels into separate shared objects.
        --config=v2             # Build TensorFlow 2.x instead of 1.x.
Preconfigured Bazel build configs to DISABLE default on features:
        --config=noaws          # Disable AWS S3 filesystem support.
        --config=nogcp          # Disable GCP support.
        --config=nohdfs         # Disable HDFS support.
        --config=nonccl         # Disable NVIDIA NCCL support.
```

Ran the following command to build
`bazel build --config=opt //tensorflow/tools/pip_package:build_pip_package`

Also tried,
`bazel build --config=opt --define=no_tensorflow_py_deps=true //tensorflow/tools/pip_package:build_pip_package`

Received error shown above. FYI, 2.3 builds fine. And I also tried a Bazel 3.7, with the same issue.

**Comment by Saduf2019** [comment]
@oracle3001 
Can you please let us know if you are using anaconda python and if yes please use normal python instead.

**Comment by oracle3001** [comment]
Yes I am using Anaconda as standard. As stated above, this has not ever been an issue with building previous versions of python.

I have tried with the stand-alone installation of python (3.8), reran` configure.py `to ensure it used this. 

```
You have bazel 3.1.0 installed.
Please specify the location of python. [Default is C:\Users\Adam\AppData\Local\Programs\Python\Python38\python.exe]:


Found possible Python library paths:
  C:\Users\Adam\AppData\Local\Programs\Python\Python38\lib\site-packages
Please input the desired Python library path to use.  Default is [C:\Users\Adam\AppData\Local\Programs\Python\Python38\lib\site-packages]

Do you wish to build TensorFlow with ROCm support? [y/N]:
No ROCm support will be enabled for TensorFlow.

Do you wish to build TensorFlow with CUDA support? [y/N]:
No CUDA support will be enabled for TensorFlow.

Please specify optimization flags to use during compilation when bazel option "--config=opt" is specified [Default is /arch:AVX]:

Would you like to override eigen strong inline for some C++ compilation to reduce the compilation time? [Y/n]:

Would you like to interactively configure ./WORKSPACE for Android builds? [y/N]:
```

Then ran 

```
bazel clean --expunge
azel build --config=opt //tensorflow/tools/pip_package:build_pip_package
```

same error as initially reported.

**Comment by oracle3001** [comment]
I have managed to fix this problem. I don't know enough about msys64 to know why this worked, but.

I reinstalled msys64 with the latest version. As stated on the TF instruction page, I ran `pacman -S git patch unzip`, but this still resulted in the reported error.

I then ran `pacman -Syuu patch`, based upon a comment I saw on another github repo where people were having issues with go related things on windows. This appears to have fixed the problem.

As I say, I don't know anything really about msys64 or why adding patch would work, but it does. Perhaps it needs adding to the TF instructions?
